documentation/Deprecation guidelines.md

```diff
@@ -0,0 +1,11 @@
+# Deprecation guidelines
+
+- Ship backwards compatible changes which include supporting both the old and new API
+- Support backwards compatibility for at least half of a major release cycle, but never more that 2 major release cycles
```

Comment:
How do you quantify a release cycle, and thus how long is half a release cycle? AFAIK we don't have a fixed period for when we release major versions.

I think it is reasonable to expect people to upgrade to the final release of a major version, fix any deprecations and then move onwards to the next major version.

Reply:
> How do you quantify a release cycle, and thus how long is half a release cycle

It’s not meant to be a quantified number, otherwise I would provide a formula 😄That’s also why these are guidelines and not rules – use best judgement in determining what is an appropriate amount of time for a deprecation.